crates/sui-indexer/src/store/indexer_store.rs

```diff
@@ -224,6 +224,24 @@ impl ObjectStore for CheckpointData {
                 _ => None,
             }))
     }
+
+    fn get_object_by_key(
+        &self,
+        object_id: &ObjectID,
+        version: VersionNumber,
+    ) -> Result<Option<sui_types::object::Object>, SuiError> {
+        Ok(self
+            .changed_objects
+            .iter()
+            .find_map(|(status, o)| match status {
+                ObjectStatus::Created | ObjectStatus::Mutated
+                    if &o.object_id == object_id && o.version == version =>
+                {
+                    o.clone().try_into().ok()
+                }
+                _ => None,
```

Comment:
Exhaustive pattern matching here would make me feel better--I don't know this code well enough to understand what other `ObjectStatus`'s are possible

Reply:
Looking at the existing status I think `Unwrapped` (?) and maybe `Wrapped` might also make sense here? But now I'm confused about what that status means. Maybe just leave this as `unimplemented!` to make the trait happy and allow indexer folks to fill it in later (assuming that doesn't cause any crashes).

Reply:
here we were trying to mimic what `get_object` above is doing, given that In fact this is an exact copy and paste of the code above.
@gegaowp @patrickkuo could you please take a look and suggest what to do?
As @sblackshear pointed out we could even leave it `unimplemented` if that is a better way to move forward

Reply:
@dariorussi indexer has a similar impl here
https://github.com/MystenLabs/sui/blob/main/crates/sui-indexer/src/store/pg_indexer_store.rs#L114
indexer_store.rs is meant to be an "interface" and we are putting all impl. in pg_indexer_store.rs

should be a quick move, otherwise feel free to leave it `todo!()` with a `MUSTFIX(gegaowp)` comment to unblock, I will do that later.